packages/builder-tools/connect-utils/vite-plugins/ph-external-packages.ts

```diff
@@ -1,119 +1,147 @@
-import type { PluginOption } from "vite";
+import type { ModuleNode, PluginOption } from "vite";
 
 /**
  * Takes a list of Powerhouse project packages, by name or path and
  * returns a js module that exports those packages for use in Connect.
  */
 function makeImportScriptFromPackages(packages: string[]) {
-  const imports: string[] = [];
-  const moduleNames: string[] = [];
-  let counter = 0;
-
-  for (const packageName of packages) {
-    const moduleName = `module${counter}`;
-    moduleNames.push(moduleName);
-    imports.push(`import * as ${moduleName} from '${packageName}';`);
-    imports.push(`import '${packageName}/style.css';`);
-    counter++;
-  }
-
-  const exports = moduleNames.map(
-    (name, index) => `{
-      id: "${packages[index]}",
-      ...${name},
-    }`,
-  );
-
-  const exportsString = exports.length
-    ? `
-        ${exports.join(",\n")}
-    `
-    : "";
-
-  const exportStatement = `export default [${exportsString}];`;
-
-  // Add HMR boundary to prevent page reload
-  const hmrCode = `
-// HMR boundary - accept updates to prevent full page reload
-if (import.meta.hot) {
-  import.meta.hot.accept();
-}`;
-
-  const fileContent = `${imports.join("\n")}\n\n${exportStatement}\n${hmrCode}`;
-
-  return fileContent;
+  return `
+   export async function loadExternalPackages() {
+      const modules = [];
+      ${packages
+        .map(
+          (pkg, index) =>
+            `try {
+          const module = await import('${pkg}');
+          await import('${pkg}/style.css');
+          modules.push({
+              id: 'module${index}',
+              ...module,
+          });
+      } catch (error) {
+          console.error("Error loading package", pkg, error);
```

Comment:
Load the external packages separately to prevent an invalid package from causing all other packages to not be loaded.